core/src/main/java/com/amplitude/core/utilities/EventsFileManager.kt

```diff
@@ -51,12 +48,12 @@ class EventsFileManager(
             // update index
             file = currentFile()
             file.createNewFile()
-            start(file)
-            newFile = true
         }
 
         var contents = ""
-        if (!newFile) {
```

Comment:
Question: Will remove the newFile flag have a concurrency issue?

Reply:
I think there is concurrency issue somewhere else since the example of customer's payload have string like [[{event}{event} that are caused by two threads writing to file. But the newFile is a variable inside a function scope and not shared by multiple threads, it's not related to concurrency issue.